File: vb.net/mvc/Controllers/TestController.vb
Imports System.Web.Mvc

Namespace Controllers
    Public Class TestController
        Inherits Controller

        ' GET: Test
        Function Index() As ActionResult
            Return View()
        End Function

        Function Login() As ActionResult
            Return View()
        End Function


        Function PostLogin(Name As String, Pwd As String) As ActionResult
            If Name = "admin" And Pwd = "123" Then
                Return Content("<script>location.href='/test/index'</script>")
            Else
                Return Content("<script>location.href='/test/login'</script>")
            End If
        End Function
    End Class
End Namespace

File: Test/WebApplication1/Controllers/TestController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace WebApplication1.Controllers
{
    public class TestController : Controller
    {
        // GET: Test
        public ActionResult Index()
        {
            return View();
        }
        [HttpPost]
        public ActionResult Index2()
        {
            return Content("<script>alert('跳转成功');location.href='/Test/Item';</script>");
            //"text/html;charset=utf-8"可不设置
        }
        public ActionResult Item()
        {
            return View();
        }
    }
}